observability/probe/prober_test.go

```diff
@@ -20,6 +20,9 @@ func TestProbeNotImplemented(t *testing.T) {
 	if !errors.IsNotImplemented(err) {
 		t.Errorf("expected probe.NotImplemented to return an error that satisfies errors.IsNotImplemented")
 	}
+	if err.Error() != "probe not implemented" {
+		t.Errorf(`expected probe.NotImplemented to string to "probe not implemented" got: %q`, err.Error())
+	}
```

Comment:
Why was this required?

Seems like we're not correctly marshalling our errors.

Reply:
The previous version was doing `return NotImplemented("probe not implemented")` which caused it to be marshaled as "probe not implemented not implemented" when looking at various logs.